addons/account/models/account_report.py

```diff
@@ -485,11 +485,21 @@ def create(self, vals_list):
         return result
 
     def write(self, vals):
-        if 'formula' not in vals:
-            return super().write(vals)
 
         self._strip_formula(vals)
 
+        if vals.get('engine') == 'tax_tags':
+            tag_name = vals.get('formula') or self.formula
+            country = self.report_line_id.report_id.country_id
+            existing_tags = self.env['account.account.tag']._get_tax_tags(tag_name, country.id)
+            if len(existing_tags) < 2:
+                tag_vals = self._get_tags_create_vals(tag_name, country.id, existing_tag=existing_tags)
+                self.env['account.account.tag'].create(tag_vals)
```

Comment:
Since this is exactly the same logic as the one in `create()`, don't you think it would be cleaner to extract it in a dedicated method? :wink: